### ISSUE TITLE
[Product] Change of status on variant selection not announced

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#463]
## Issue summary

On product pages and featured product section, the status of the product may change as the user selects a variant. For example, selection one size or another reveals if the product is on sale or sold out. This status is only conveyed visually. Screen reader users may miss this update which could lead to a confusing or frustrating user experience (as described in https://github.com/Shopify/dawn/issues/1.)

<details>
  <summary>Screenshots</summary>

![Gif of product page. Size variant selector moves between and selects shirt sizes. Sale price and sold out status text is updated above the selectors.](https://user-images.githubusercontent.com/1392632/130497207-5bc7c07c-eaba-416b-8393-32a2fe32f5c1.gif)

</details>

#### HTML

```html
<div
  class="no-js-hidden"
  id="price-template--15004892561579__1629739262600943ca"
  …
>
  <div class="price price--large price--on-sale">
    <!-- Price content… -->
    <span class="badge price__badge-sale color-accent-2" aria-hidden="true">
      Sale
    </span>
    <span class="badge price__badge-sold-out color-inverse" aria-hidden="true">
      Sold out
    </span>
  </div>
</div>
```

## Steps to reproduce

1. Go to a Dawn store, Product page with any supported browser
2. With an active screen reader, navigate to the variant selector and activate variants via arrow keys to reveal status content

### Expected

- Price and status content to be announced by the screen reader.

### Actual

- Price and status is not announced.

### HTML structure…

1. Add the `role` attribute to the `div#price-template` content container, setting its value to `status`. This will create a live region for screen readers to announce changed content.
2. Remove the `aria-hidden` attributes from the `span.badge` elements in order for screen readers to include this information.

#### HTML

```html
<div
  class="no-js-hidden"
  id="price-template--15004892561579__1629739262600943ca"
  …
  role="status"
>
  <div class="price price--large price--on-sale">
    <!-- Price content… -->
    <span class="badge price__badge-sale color-accent-2">
      Sale
    </span>
    <span class="badge price__badge-sold-out color-inverse">
      Sold out
    </span>
  </div>
</div>
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **WCAG Principal:** Robust
- **WCAG SC:** [4.1.3 Status Messages](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html)
- **Severity:** High
- **Effort:** Low

**Comment by svinkle** [comment]
Increasing the `Severity` as this is critical information to share with screen reader users.
